### PR TITLE
ncm-authconfig: switch to CCM::TextRender and TT unittests

### LINKED ISSUE #440: ncm-authconfig: switch to CAF::TextRender and unittests
To avoid errors on EL5 like
```
#   Failed test 'No errors in rendering the template'
#   at src/test/perl/sssd.t line 51.
#          got: undef
#     expected: ''
```
and use proper tests for the sssd TT files

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #440

### CODE REVIEW

File: ncm-authconfig/src/main/pan/components/authconfig/config.pan [hunk 1]
```diff
@@ -3,7 +3,33 @@
...
+
+"safemode" ?= false;
+
+"useshadow" ?= true;
+"usecache" ?= true;
+
+"usemd5" ?= true;
+"passalgorithm" ?= {
+    if (value("/software/components/${project.artifactId}/usemd5")) {
+        v = "md5";
```
Comment (ned21) [inline ncm-authconfig/src/main/pan/components/authconfig/config.pan]: The extra v variable is redundant:
```
passalgorithm ?= {
  if(value("...")) {
    "md5";
  } else {
    # Fall back to the most portable option
    "descrypt";
  };
};

**Comment by ned21** [comment]
Do commits 3cb9c14 and ee1e53c not cancel each other out and can both just be deleted?

File: ncm-authconfig/src/main/perl/authconfig.pm [hunk 2]
```diff
@@ -2,28 +2,25 @@
...
-
 package NCM::Component::authconfig;
 
 use strict;
 use NCM::Component;
-use CAF::Process;
 use vars qw(@ISA $EC);
 @ISA = qw(NCM::Component);
 $EC=LC::Exception::Context->new->will_store_all;
+
```
Comment (ned21) [inline ncm-authconfig/src/main/perl/authconfig.pm]: This looks to support NoAction mode (as in, it only use CAF libraries to do stuff) so can you mark it as NoActionSupported please?

**Comment by stdweird** [comment]
@ned21 remarks processed

**Comment by ned21** [comment]
LGTM (assuming merge build tests pass!)

**Comment by stdweird** [comment]
part of https://github.com/quattor/release/issues/85; with PR merged, tests pass on el5

**Pull request merged**